lib/src/main/java/com/shopify/checkoutsheetkit/BaseWebView.kt

```diff
@@ -65,6 +70,32 @@ internal abstract class BaseWebView(context: Context, attributeSet: AttributeSet
                 super.onProgressChanged(view, newProgress)
                 getEventProcessor().updateProgressBar(newProgress)
             }
+            override fun onPermissionRequest(request: PermissionRequest) {
+                request.resources?.forEach { resource ->
+                    if (resource == PermissionRequest.RESOURCE_VIDEO_CAPTURE) {
+                        if (!hasCameraPermission()) {
+                            requestPermissions()
+                            request.deny()
+                        } else {
+                            request.grant(request.resources)
+                        }
+                    }
+                }
+            }
+
+            private fun hasCameraPermission() =
+                ActivityCompat.checkSelfPermission(
+                    context,
+                    Manifest.permission.CAMERA
+                ) == PackageManager.PERMISSION_GRANTED
+
+            private fun requestPermissions() {
+                ActivityCompat.requestPermissions(
+                    context as ComponentActivity,
+                    arrayOf(Manifest.permission.CAMERA),
+                    CAMERA_PERMISSION_REQUEST
+                )
+            }
```

Comment:
Could we make this generic? 🤔 
```suggestion
            private fun hasPermission(Permission permission) =
                ActivityCompat.checkSelfPermission(
                    context,
                    permission
                ) == PackageManager.PERMISSION_GRANTED

            private fun requestPermissions(Permission permission) {
                ActivityCompat.requestPermissions(
                    context as ComponentActivity,
                    arrayOf(permission),
                    1
                )
            }
```